CRM/Contribute/Form/Search.php

```diff
@@ -390,6 +390,11 @@ public function fixFormValues() {
       $this->_defaults['contribution_status_id'] = array($status => 1);
     }
 
+    $pcpid=CRM_Utils_Request::retrieve('pcpid', 'Positive', $this);
+    if ($pcpid) {
```

Comment:
I think there is a misunderstanding. 
I modified the fixFormValues method, which is only executed when $this->_force is true.
Check:
`  public function fixFormValues() {

    if (!$this->_force) {

      return;

    }`

$this->_force is set from the query  string :
`    /*

     * we allow the controller to set force/reset externally, useful when we are being

     * driven by the wizard framework

     */
    $this->_reset = CRM_Utils_Request::retrieve('reset', 'Boolean', CRM_Core_DAO::$_nullObject);

    $this->_force = CRM_Utils_Request::retrieve('force', 'Boolean', $this, FALSE);

    $this->_limit = CRM_Utils_Request::retrieve('limit', 'Positive', $this);`

and its purpose is driving the Search process when NO form has been submitted.

Original method allowed for examp using url like

> http://localhost/drupal7/civicrm/contribute/search?force=1&cid=_contact_id_>
in order to obtain all Contributions for a specific user
or
> http://localhost/drupal7/civicrm/contribute/search?force=1&pid=_contribution_page_id_>
in order to obtain all Contributions for a specific Contribution Page

I knew that contribution_pcp_made_through_id is an array and sued it as an array, but it was for me an OUTPUT field.

My input is a query string parameter called pcpid in order to obtain all Contributions for a specific Personal Campaign. This field was not intended to be an array because I was tasked to run this search simply forcing one Personal Campaign.

Your code was breaking for the url : http://localhost/drupal7/civicrm/contribute/search?force=1&pcpid=1 while mine works for both
http://localhost/drupal7/civicrm/contribute/search?force=1&pcpid=1
and
http://localhost/drupal7/civicrm/contribute/search?force=1&pcpid[]=1

I hope i succesfully made it clear this time. I somewhat assumed that who was to review this code knew the actual assignment that raised it's implementation.

Please check the last commit
Best regards